ydb/core/tx/schemeshard/schemeshard_validate_ttl.cpp

```diff
@@ -58,8 +58,14 @@ bool ValidateTtlSettings(const NKikimrSchemeOp::TTTLSettings& ttl,
             return false;
         }
 
+        if (enabled.TiersSize() > 1 || (enabled.TiersSize() == 1 && !enabled.GetTiers(0).HasDelete())) {
```

Comment:
Can't quite understand this check: "there should be a single tier only and it should be a Delete one"? If so, then why not to check exactly for that?
And what about several tiers, all with Delete action? -- technically its possible to construct a message that way and it would be rejected by this check but with the misleading error message.

Reply:
And why this check is made outside of `GetExpireAfter()`?

Reply:
> And what about several tiers, all with Delete action?

Only the last tier is allowed to contain Delete action. I added validation of this case here to return a better error message.